src/client/pps.go

```diff
@@ -232,18 +221,84 @@ func (c APIClient) InspectJob(jobID string, blockState bool, full ...bool) (*pps
 	return jobInfo, grpcutil.ScrubGRPC(err)
 }
 
-// InspectJobOutputCommit returns info about a job that created a commit.
-// blockState will cause the call to block until the job reaches a terminal state (failure or success).
-func (c APIClient) InspectJobOutputCommit(repoName, branchName, commitID string, blockState bool) (*pps.JobInfo, error) {
-	jobInfo, err := c.PpsAPIClient.InspectJob(
-		c.Ctx(),
-		&pps.InspectJobRequest{
-			OutputCommit: NewCommit(repoName, branchName, commitID),
-			BlockState:   blockState,
-		})
+// BlockJob is a blocking version on InspectJob that will block
+// until the job has reached a terminal state.
+func (c APIClient) BlockJob(pipelineName string, jobID string, full ...bool) (*pps.JobInfo, error) {
+	req := &pps.InspectJobRequest{
+		Job:        NewJob(pipelineName, jobID),
+		BlockState: true,
+	}
+	if len(full) > 0 {
+		req.Full = full[0]
+	}
+	jobInfo, err := c.PpsAPIClient.InspectJob(c.Ctx(), req)
 	return jobInfo, grpcutil.ScrubGRPC(err)
 }
 
+func (c APIClient) inspectJobset(id string, block bool, full bool, cb func(*pps.JobInfo) error) (retErr error) {
+	defer func() {
+		retErr = grpcutil.ScrubGRPC(retErr)
+	}()
+	req := &pps.InspectJobsetRequest{
+		Jobset: NewJobset(id),
+		Block:  block,
+		Full:   full,
+	}
+	client, err := c.PpsAPIClient.InspectJobset(c.Ctx(), req)
+	if err != nil {
+		return err
+	}
+	for {
+		ci, err := client.Recv()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				return nil
+			}
+			return err
+		}
+		if err := cb(ci); err != nil {
+			if errors.Is(err, errutil.ErrBreak) {
+				return nil
+			}
+			return err
+		}
+	}
+}
+
+func (c APIClient) InspectJobset(id string, full ...bool) ([]*pps.JobInfo, error) {
+	isFull := false
+	if len(full) > 0 {
+		isFull = full[0]
+	}
+	result := []*pps.JobInfo{}
+	if err := c.inspectJobset(id, false, isFull, func(ji *pps.JobInfo) error {
+		result = append(result, ji)
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func (c APIClient) BlockJobsetAll(id string, full ...bool) ([]*pps.JobInfo, error) {
+	isFull := false
+	if len(full) > 0 {
+		isFull = full[0]
+	}
+	result := []*pps.JobInfo{}
+	if err := c.inspectJobset(id, true, isFull, func(ji *pps.JobInfo) error {
```

Comment:
Same as what I mentioned above, it would be better for this to call `BlockJobset`. Also, I wouldn't be against making full required. I'm on board with moving away from variadic parameters for single optional parameters. Having the pattern of required parameters or functional options makes sense to me.

Reply:
I don't think we were actually using `Full=true` anyway - removed and if you really need it you can use the lower-level interface (or we can go ahead with a more powerful functional interface).